.github/workflows/main.yml

```diff
@@ -16,18 +16,25 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ['3.8', '3.9', '3.10', '3.11']
-      fail-fast: False
+      fail-fast: false
 
     steps:
     - uses: actions/checkout@v3
-    - name: Set up Python ${{ matrix.python-version }}
+    - name: Set up python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Install
+    - name: Install dependencies
       run: make ci-install
+      shell: bash
+
+    - name: Check linting and type annotations
+      continue-on-error: true
```

Comment:
Well this apparently doesn't fail the CI if this step fails, which is not great :(

Reply:
Adding this seems to work (from https://stackoverflow.com/a/73015952)
```
      if: success() || failure()
```